src/lib/Semantics.ml

```diff
@@ -857,11 +854,20 @@ and plug_into ~style sp con =
   | `Done -> ret @@ `Reduce res
   | `Reduce res -> ret @@ `Reduce res
 
-and whnf_hd ?(style = default_whnf_style) hd =
+and should_unfold_symbol style sym =
+  match style with
+  | `UnfoldNone -> false
+  | `UnfoldAll -> true
+  | `Veil veil ->
+    match Veil.policy sym veil with
```

Comment:
I wonder if it makes sense to update the veil so that it only unfold <= N times or maybe only under <= M binders LOL.

Reply:
LOL. I am open to many different Cool ideas for veil policies haha. Though I think that one may be a little fragile.

Anyway, we can make a Cool Mini-Language for unfolding if we are board ;-)

Reply:
Let's leave my Immortal Thought in another thread...